tenant/config/crds/tenancy_v1alpha1_tenant.yaml

```diff
@@ -28,6 +28,11 @@ spec:
           type: object
         spec:
           properties:
+            requireNamespacePrefix:
+              description: If set to True, all the namespaces belong to the tenant
+                are requried to have TenantAdminNamespaceName as name prefix. By default,
+                namespace prefix is not required.
+              type: boolean
```

Comment:
General question. Should this be an enum instead?

Maybe some site wants to use the namespace as a suffix instead of a prefix, or some other idea we haven't thought of yet.

namespacePattern: [ TenantNamespace, NamespaceTenant, Namespace ] ?

Reply:
I‘d rather not go with this far if a boolean can serve 90% use cases.

Reply:
@Fei-Guo I agree with keeping it simple, but it does bring up a good point. Most generated items in k8s are suffixs rather than prefixes. A pro for a prefix is you get sorting when doing a list, however, a suffix does hide a redundant item and follows the precedent set by deployments -> pods, cronjobs -> jobs, etc.